tests/ssg_test_suite/oscap.py

```diff
@@ -103,75 +162,108 @@ def run_rule(domain_ip,
                                 '{0}-{1}-{2}'.format(rule_id,
                                                      script_name,
                                                      stage))
-    formatting['report'] = LogHelper.find_name(report_path, '.html')
+    arf_path = os.path.join(LogHelper.LOG_DIR,
+                            '{0}-initial-arf.xml'.format(profile))
+
     verbose_path = LogHelper.find_name(verbose_path, '.verbose.log')
 
-    command = shlex.split(('oscap-ssh root@{domain_ip} 22 xccdf eval '
-                           '--benchmark-id {benchmark_id} '
-                           '--profile {profile} '
-                           '--progress --oval-results '
-                           '--rule {rule_id} '
-                           '--report {report} '
-                           '--verbose DEVEL '
-                           '{rem} '
-                           '{datastream}').format(**formatting))
-    logging.debug('Running ' + ' '.join(command))
+    if stage is 'initial' and ansible:
+        formatting['arf'] = "--results-arf {}".format(arf_path)
+    else:
+        formatting['arf'] = ""
 
     success = True
-    # check expected return code
+    # Set expected return code.
     expected_return_code = _CONTEXT_RETURN_CODES[context]
-    try:
-        with open(verbose_path, 'w') as verbose_file:
-            output = subprocess.check_output(command, stderr=verbose_file)
 
-    except subprocess.CalledProcessError, e:
-        if e.returncode != expected_return_code:
+    if stage is 'remediation' and ansible:
```

Comment:
This `if` block seems is too much, imho - can you carve it out as separate function, same with bash? So the `run_rule` will be easier to read overall.